src/directions/directions_renderer.ts

```diff
@@ -177,75 +180,69 @@ export class MigrationDirectionsRenderer {
-        }
-
-        maplibreMap.addLayer({
-          id: routeId,
-          type: "line",
-          source: routeId,
-          layout: {
-            "line-join": "round",
-            "line-cap": "round",
-            visibility: this.#polylineOptions && this.#polylineOptions.visible == false ? "none" : "visible",
-          },
-          paint: paintOptions,
-        });
 
-        this.#legRenderIds.push(routeId);
+    // Draw single LineString for the entire route
+    // TODO: Detect geometry type instead of just doing LineString
+    if (this.#suppressPolylines === false) {
+      const routeId = `directions-renderer-${this.rendererIndex}-route-${this.#routeIndex}`;
+      maplibreMap.addSource(routeId, {
+        type: "geojson",
+        data: {
+          type: "Feature",
+          properties: {},
+          geometry: {
+            type: "LineString",
+            coordinates: route.overview_path.map((coord) => [coord.lng(), coord.lat()]),
+          },
+        },
+      });
+      // 8 weight, 0.5 opacity, "#73B9FF" color for default, 3 weight, 1 opacity, "Black" color used when one property is set
+      const paintOptions = {};
+      if (this.#polylineOptions) {
+        paintOptions["line-color"] = this.#polylineOptions.strokeColor ? this.#polylineOptions.strokeColor : "Black";
+        paintOptions["line-width"] = this.#polylineOptions.strokeWeight ? this.#polylineOptions.strokeWeight : 3;
+        paintOptions["line-opacity"] = this.#polylineOptions.strokeOpacity ? this.#polylineOptions.strokeOpacity : 1;
+      } else {
+        // default line
+        paintOptions["line-color"] = DEFAULT_LINE_COLOR;
+        paintOptions["line-width"] = DEFAULT_LINE_WIDTH;
+        paintOptions["line-opacity"] = DEFAULT_LINE_OPACITY;
       }
 
-      // Add markers for the start location of the current leg
-      if (this.#suppressMarkers === false) {
-        const startLocation = leg.start_location;
-
-        const startMarkerOptions =
-          this.#markerOptions === undefined
-            ? { label: String.fromCharCode(ASCII_CODE_A + i) }
-            : structuredClone(this.#markerOptions);
-        startMarkerOptions.position = startLocation;
-        startMarkerOptions.map = this.#map;
-        const startMarker = new MigrationMarker(startMarkerOptions);
-        this.#markers.push(startMarker);
-      }
+      // Add the route layer
+      maplibreMap.addLayer({
+        id: routeId,
+        type: "line",
+        source: routeId,
+        layout: {
+          "line-join": "round",
+          "line-cap": "round",
+          visibility: this.#polylineOptions && this.#polylineOptions.visible == false ? "none" : "visible",
+        },
+        paint: paintOptions,
+      });
+
+      this.#legRenderIds.push(routeId);
 
       // TODO: Add default info windows once location information is passed into route result
     }
 
-    // Add final marker for end location of enture route
+    // Add markers (if not suppressed)
```

Comment:
The old code was adding markers for each leg of the trip. The new code looks like it is only adding start and end markers for the entire route. Is this change intentional?

Reply:
OK, are you saying that with this change, the Google directions example and the Amazon Location directions example will now look the same, and they didn't look the same before? Or are you saying that they will look different now, but we want them to look different?